src/index.ts

```diff
@@ -60,44 +145,23 @@ document.getElementById('clear-button')!.onclick = function() {
     redraw();
 };
 
-const parameterSlider = document.getElementById('parameter-slider') as HTMLInputElement;
-parameterSlider.max = (fftSize - 1).toString();
-parameter = parameterSlider.valueAsNumber;
-parameterSlider.oninput = function() {
-    parameter = parameterSlider.valueAsNumber;
-    redraw();
-};
-const complexityNumber = document.getElementById('complexity-number') as HTMLInputElement;
-complexityNumber.max = (fftSize - 1).toString();
-complexity = complexityNumber.valueAsNumber;
-complexityNumber.oninput = function() {
-    complexity = complexityNumber.valueAsNumber;
-    redraw();
-};
-const complexityCircles = document.getElementById('complexity-circles-check') as HTMLInputElement;
-circles = complexityCircles.checked;
-complexityCircles.oninput = function() {
-    circles = complexityCircles.checked;
-    redraw();
-};
+document.getElementById('save-button')!.onclick = setLocation;
 
 function magnitude(x: number, y: number) { return Math.sqrt(x * x + y * y); }
 
 function lerp(first: number, second: number, t: number) { return first + (second - first) * t; }
 
-function addPoint(x: number, y: number) {
+function addPoint(x: number, y: number, draw = true) {
     if (points.length === 0) {
         points.push({ x, y, segmentLength: 0 });
+        points.push({ x, y, segmentLength: 0 });
     } else {
-        const previousPoint = points[Math.max(0, points.length - 2)];
+        const previousPoint = points[points.length - 2];
         const segmentLength = magnitude(x - previousPoint.x, y - previousPoint.y);
         unclosedLength += segmentLength;
-
-        const addedPoint = { x, y, segmentLength };
-        points.splice(points.length - 1, 0, addedPoint);
-
-        const startAndEndPoint = points[points.length - 1];
-        startAndEndPoint.segmentLength = magnitude(startAndEndPoint.x - x, startAndEndPoint.y - y);
+        points[points.length - 1] = { x, y, segmentLength };
+        const firstPoint = points[0];
+        points.push({ x: firstPoint.x, y: firstPoint.y, segmentLength: magnitude(firstPoint.x - x, firstPoint.y - y) });

```

Comment:
Why these changes and the ones in `samplePathIntoInput` (https://github.com/jnm2/epic/pull/4/commits/c76407139c1d06828882d075425ab146e9c40c0c)? I think it is undoing https://github.com/jnm2/epic/commit/370c1ba7395ece25708e3f54483be247599131fc.